source/solvers/vof_algebraic_interface_reinitialization.cc

```diff
@@ -0,0 +1,827 @@
+
+  // Get residual conditions
+  const double absolute_residual =
+    this->simulation_parameters.vof_subequations_linear_solvers
+      .at(VOFSubequationsID::algebraic_interface_reinitialization)
+      .minimum_residual;
+  const double relative_residual =
+    this->simulation_parameters.vof_subequations_linear_solvers
+      .at(VOFSubequationsID::algebraic_interface_reinitialization)
+      .relative_residual;
+
+  // Set linear solver tolerance
+  const double linear_solver_tolerance =
+    std::max(relative_residual * this->system_rhs.l2_norm(), absolute_residual);
+
+  if (verbose)
+    {
+      this->pcout << "  -Tolerance of iterative solver is: "
+                  << linear_solver_tolerance << std::endl;
+    }
+
+  // ILU preconditioner
+  const double ilu_fill =
+    this->simulation_parameters.vof_subequations_linear_solvers
+      .at(VOFSubequationsID::algebraic_interface_reinitialization)
+      .ilu_precond_fill;
+  const double ilu_atol =
+    this->simulation_parameters.vof_subequations_linear_solvers
+      .at(VOFSubequationsID::algebraic_interface_reinitialization)
+      .ilu_precond_atol;
+  const double ilu_rtol =
+    this->simulation_parameters.vof_subequations_linear_solvers
+      .at(VOFSubequationsID::algebraic_interface_reinitialization)
+      .ilu_precond_rtol;
+  TrilinosWrappers::PreconditionILU::AdditionalData preconditionerOptions(
+    ilu_fill, ilu_atol, ilu_rtol, 0);
+
+  TrilinosWrappers::PreconditionILU ilu_preconditioner;
+
+  ilu_preconditioner.initialize(this->system_matrix, preconditionerOptions);
+
+  GlobalVectorType completely_distributed_solution(this->locally_owned_dofs,
+                                                   mpi_communicator);
+
+  SolverControl solver_control(
+    this->simulation_parameters.vof_subequations_linear_solvers
+      .at(VOFSubequationsID::algebraic_interface_reinitialization)
+      .max_iterations,
+    linear_solver_tolerance,
+    true,
+    true);
+
+  TrilinosWrappers::SolverGMRES::AdditionalData solver_parameters(
+    false,
+    this->simulation_parameters.vof_subequations_linear_solvers
+      .at(VOFSubequationsID::algebraic_interface_reinitialization)
+      .max_krylov_vectors);
+
+  TrilinosWrappers::SolverGMRES solver(solver_control, solver_parameters);
+
+  solver.solve(this->system_matrix,
+               completely_distributed_solution,
+               this->system_rhs,
+               ilu_preconditioner);
+
+  if (verbose)
+    {
+      this->pcout << "  -Iterative solver took: " << solver_control.last_step()
+                  << " steps to reach a residual norm of "
+                  << solver_control.last_value() << std::endl;
+    }
+
+  // Update constraints and newton vectors
+  constraints_used.distribute(completely_distributed_solution);
+  this->newton_update = completely_distributed_solution;
+}
+
+
+template <int dim>
+void
+VOFAlgebraicInterfaceReinitialization<dim>::solve(
+  const bool & /*is_post_mesh_adaptation = false*/)
```

Comment:
```suggestion
  const bool & /*is_post_mesh_adaptation = false*/)
```
Can't this bool be removed?